test/test_configs.py

```diff
@@ -436,6 +442,69 @@ def test_redq(self, pixels, independent, continuous):
         qvalue(tensordict)
 
 
+@pytest.mark.skipif(not _has_hydra, reason="No hydra found")
+class TestExplorationConfigs:
+    from torchrl.modules.models.models import _LAYER_CLASS_DICT
+    from torchrl.modules.tensordict_module.exploration import (
+        EGreedyWrapper,
+        AdditiveGaussianWrapper,
+        OrnsteinUhlenbeckProcessWrapper,
+    )
+
+    wrapper_map = {
+        "e_greedy": EGreedyWrapper,
+        "additive_gaussian": AdditiveGaussianWrapper,
+        "ou_process": OrnsteinUhlenbeckProcessWrapper,
+    }
+
+    @pytest.mark.parametrize("network", ["linear", "noisy_linear"])
+    @pytest.mark.parametrize(
+        "wrapper", [None, "e_greedy", "additive_gaussian", "ou_process"]
+    )
+    def test_exploration(self, network, wrapper):
+        additional_config = [
+            "env=cartpole",
+            "transforms=state",
+            "model=sac/discrete",
+            "network=sac/independent_state",
+        ]
+        exploration_config = []
+        if wrapper is not None:
+            exploration_config += [
+                f"exploration={wrapper}",
+                f"exploration.network={network}",
+            ]
+        else:
+            exploration_config += [f"exploration={network}"]
+
+        cfg = hydra.compose("config", overrides=exploration_config + additional_config)
+        env = instantiate(cfg.env)
+        transforms = [instantiate(transform) for transform in cfg.transforms]
+        for t in transforms:
+            env.append_transform(t)
+        model_params = instantiate(cfg.model)
+        net_partial = instantiate(cfg.network)
+        actor, qvalue, value = make_model_sac(net_partial, model_params, env)
+        actor(env.reset())
+        assert actor.module.module.layer_class is self._LAYER_CLASS_DICT[network]
```

Comment:
This whole block is orthogonal to the exploration. 
We can think about it -- perhaps it's better like this -- we might miss incompatibilities with mocking classes.
Otherwise I can help you to make one or two mocking classes to have a test that does not rely on other configs.

Reply:
Recreated environment from printout, rather than relying on config file